src/antsibull_core/args.py

```diff
@@ -16,27 +16,39 @@ class InvalidArgumentError(Exception):
     """A problem parsing or validating a command line argument."""
 
 
-def get_toplevel_parser(package, **kwargs) -> argparse.ArgumentParser:
+def get_toplevel_parser(
+    package: str,
+    *,
+    package_version: str | None = None,
+    program_name: str | None = None,
+    **kwargs,
+) -> argparse.ArgumentParser:
     """
     Create a toplevel argument parser with options common across all scripts.
 
     :arg package: The Python package containing this CLI program.
+    :arg package_version: If provided, use this instead of the version for ``package``
+        provided by ``importlib.metadata.version()``.
+    :arg program_name: If provided, show a more concrete description for this program.
     :args kwargs: This function takes any keyword arguments and passes them directly on to
         the :class:`argparse.ArgumentParser` constructor.
     :returns: :class:`argparse.ArgumentParser` with common script arguments added.
     """
-    try:
-        version = metadata.version(package)
-    except metadata.PackageNotFoundError:
-        # If there's no metadata foun, assume we're running from source
-        version = "source"
+    if package_version is None:
+        try:
+            package_version = metadata.version(package)
+        except metadata.PackageNotFoundError:
+            # If there's no metadata foun, assume we're running from source
```

Comment:
This was here before, but I guess we can fix it as part of this PR.

```suggestion
            # If there's no metadata found, assume we're running from source
```